core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -309,7 +308,7 @@ object ConsoleConsumer extends Logging {
       formatterArgs.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, valueDeserializer)
     }
 
-    formatter.init(formatterArgs)
+    formatter.configure(formatterArgs.asScala.asJava)
```

Comment:
`asScala.asJava` looks a bit weird here. I understand that you are using this trick to convert the Properties to a Scala Map and then a Java Map. How about loading `formatterArgs` as a Map directly?

Reply:
This trick is used in a couple of other places. I took a quick look at switching to a `Map` directly but it's used in a bunch of places so ended up being a larger change. I'd prefer keeping this PR focussed on the new formatter and maybe do some refactoring in a follow up PR. WDYT?

Reply:
I was thinking about adding a second utility method `CommandLineUtils.parseKeyValueArgsAsMap` and keep the existing one unchanged. We can do this as a follow up for sure.